scala/README.md

```diff
@@ -534,22 +620,73 @@ Tips & Tricks
 
     ```scala
     // Value classes
-    case class Email(email: String) extends AnyVal // no extra object allocation at runtime
+    case class Email(email: String) extends AnyVal // No extra object allocation at runtime
 
-    val email = Email("no@one.com") // never gets mixed with regular strings
+    val email = Email("joe@example.com") // Never gets mixed with other strings
 
     // Extension methods
     implicit class IntOps(val n: Int) extends AnyVal {
       def stars = "*" * n
     }
 
-    5.stars // equivalent to a static method call, no implicit conversion actually takes place
+    5.stars // Equivalent to a static method call, no implicit conversion actually takes place
     ```
 
 0. Leverage parallel collections, use `.par` judiciously.
 
 0. Use the `@tailrec` annotation to ensure the compiler can recognize a recursive method is tail-recursive.
 
+0. You can use the `@scala.beans.BeanProperty` and `@BooleanBeanProperty` annotations to automatically generate JavaBeans style getter and setter methods.
+
+0. Use this syntax to pass a sequence as parameter to a variable length argument list method:
+
+    ```scala
+    def foo(args: Int*) = ???
+
+    val seq = Seq(1, 2, 3)
+
+    foo(seq:_*)
+    ```
+
+0. There are some really neat tricks we can do with pattern matching:
+
+    ```scala
+    val tuple = ("foo", 1, false)
+    val (x, y, z) = tuple
+    assert((x, y, z) == (("foo", 1, false)))
```

Comment:
> It is a minor annoyance

Actually, I take it back. It is no annoyance at all. It is a great benefit: it discourages using tuples as method parameters.

To understand what is happening, you have to desugar the expression:

```scala
val tuple = (1, 2)

tuple == (1, 2)
// desugars to
tuple.==(1, 2) // calling equals with two parameters: the integers `1` and `2`

tuple == ((1, 2))
// desugars to
tuple.==((1, 2)) // calling equals with one parameter: the tuple `(1, 2)`
```

@matthandlersux  @a-wortman